src/main/java/com/bobocode/repo/ApplicationRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.bobocode.repo;
+
+import com.bobocode.domain.Application;
+import com.bobocode.enums.Status;
+import java.math.BigDecimal;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ApplicationRepository extends JpaRepository<Application, Long> {
+
+  Optional<Application> findFirstByStatusAndCreditAmountUSDBetweenOrderByCreatedAt(Status status,
```

Comment:
@kebann I'd say that it's a bit too long :)

Reply:
Agreed. 
I wanted to see if the derived query methods would work. 

Do you suggest creating a custom method with `@Query` instead?

Reply:
For now, just wrapped into the default method.
I see that @Query doesn't support limiting results, so it must be a custom implementation with EntityManager

Reply:
added the custom repo impl